### PR TITLE
Implemented Scale

### LINKED ISSUE #3395: Implement Scale Intrinsic Function
```
$ cat a.f90
program test_scale
    real :: x = 178.1387e-4
    integer :: i = 5
    print *, scale(x,i)
end program test_scale

$ gfortran a.f90 && ./a.out
  0.570043862 

$ lfortran a.f90
semantic error: Function 'scale' not found or not implemented yet (if it is intrinsic)
```
Implement scale intrinsic function
Refer : https://gcc.gnu.org/onlinedocs/gfortran/SCALE.html

### PR DESCRIPTION
**Pull request by parth121101**
Fix #3395

### CODE REVIEW

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -723,6 +723,7 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
         {"ishft", {IntrinsicSignature({"i", "shift"}, 2, 2)}},
         {"floor", {IntrinsicSignature({"a", "kind"}, 1, 2)}},
         {"ceiling", {IntrinsicSignature({"a", "kind"}, 1, 2)}},
+        {"scale", {IntrinsicSignature({"a", "kind"}, 2, 2)}},
```
Comment (HarshitaKalani) [inline src/lfortran/semantics/ast_common_visitor.h:726]:
```suggestion
        {"scale", {IntrinsicSignature({"X", "I"}, 2, 2)}},
```

File: src/libasr/pass/intrinsic_functions.h [hunk 2]
```diff
@@ -794,6 +795,34 @@ namespace Radix {
 
 }  // namespace Radix
 
+namespace Scale {
+    static ASR::expr_t *eval_Scale(Allocator &al, const Location &loc,
+            ASR::ttype_t* arg_type, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+        double value_X = ASR::down_cast<ASR::RealConstant_t>(expr_value(args[0]))->m_r;
+        int64_t value_I = ASR::down_cast<ASR::IntegerConstant_t>(expr_value(args[1]))->m_n;
+        double result = value_X * std::pow(2, value_I);
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_functions.h:803]: SCALE(X,I) returns X * RADIX(X)**I

Wouldn't it be better to call Radix(X) only?

File: src/libasr/pass/intrinsic_functions.h [hunk 3]
```diff
@@ -794,6 +795,34 @@ namespace Radix {
...
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("");
+        fill_func_arg("x", arg_types[0]);
+        fill_func_arg("y", arg_types[1]);
+        auto result = declare(fn_name, return_type, ReturnVar);
+        /*
+        * r = scale(x, y)
+        * r = x * 2**y
+        */
+        body.push_back(al, b.Assignment(result, r64Mul(args[0], iPow(i(2, arg_types[1]), args[1], arg_types[0]))));
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_functions.h:818]:
```suggestion
        body.push_back(al, b.Assignment(result, r_tMul(args[0], iPow(i(2, arg_types[1]), args[1], arg_types[0]), arg_types[0])));
```

Comment (HarshitaKalani) [reply src/libasr/pass/intrinsic_functions.h:818]: And define `r_tMul` in `asr_builder.h` file.
```
#define r_tMul(left, right, t) EXPR(ASR::make_RealBinOp_t(al, loc, left, \
        ASR::binopType::Mul, right, t, nullptr))
```

File: integration_tests/intrinsics_137.f90 [hunk 4]
```diff
@@ -0,0 +1,6 @@
+program intrinsic_137
+    real :: x = 178.1387e-4
+    integer :: i = 5
+    print *, scale(x,i)
+    print *, scale(178.1387e-4,5)
+end program
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_137.f90:6]: Can you add error stop conditions to make it more robust?

**Review by HarshitaKalani — APPROVED** [review]
LGTM

**Review by certik — APPROVED** [review]
Looks great. The radix() usage can be done later.

**Comment by certik** [comment]
Can you please resolve the conflicts?

**Comment by parth121101** [comment]
> Can you please resolve the conflicts?

Sure @certik , I will resolve the conflicts.

**Review by certik — APPROVED** [review]

**Pull request merged**